test/model-method.js

```diff
@@ -101,3 +101,186 @@ describe('ModelMethod', function() {
+            expect(methods).to.have.property('testMethod');
+            expect(methods).to.not.have.property('prototype.testMethod');
+            expect(methods.testMethod).to.have.property('isStatic', true);
+            done();
+          });
+        });
+      });
+  });
+
+  it('correctly adds prototype method', function(done) {
+    ModelMethod.create(
+      {
+        modelId: userModel.id,
+        name: 'testMethod',
+        isStatic: false
+      },
+      function(err) {
+        if (err) return done(err);
+        userModel.methods(function(err, list) {
+          if (err) return done(err);
+          expect(list).to.have.length(1);
+          expect(list[0]).to.have.property('name', 'testMethod');
+          expect(list[0]).to.have.property('isStatic', false);
+
+          var cfg = new ConfigFile({ path: 'server/models/user.json' });
+          cfg.load(function(err) {
+            if (err) return done(err);
+            var methods = cfg.data.methods;
+            expect(methods).to.be.an('object');
+            expect(methods).to.have.property('testMethod');
+            expect(methods).to.not.have.property('prototype.testMethod');
+            expect(methods.testMethod).to.have.property('isStatic', false);
+            done();
+          });
+        });
+      });
+  });
+});
+
+describe('ModelMethod - Loopback 3.0', function() {
+  var userModel;
+
+  beforeEach(givenLB3Workspace);
+  
+  beforeEach(function(done) {
+    ModelDefinition.create(
+      {
+        name: 'user',
+        facetName: 'server'
+      },
+      function(err, result) {
+        if (err) return done(err);
+        userModel = result;
+        done();
+      });
+  });
+ 
+  it('add static method without isStatic flag to method definition', function(done) {
+    ModelMethod.create(
+      {
+        modelId: userModel.id,
+        name: 'testMethod',
+        isStatic: true
+      },
+      function(err) {
+        if (err) return done(err);
+        userModel.methods(function(err, list) {
+          if (err) return done(err);
+          expect(list).to.have.length(1);
+          expect(list[0]).to.have.property('name', 'testMethod');
+          expect(list[0]).to.not.have.property('isStatic');
+
+          var cfg = new ConfigFile({ path: 'server/models/user.json' });
+          cfg.load(function(err) {
+            if (err) return done(err);
+            var methods = cfg.data.methods;
+            expect(methods).to.be.an('object');
+            expect(methods).to.have.property('testMethod');
+            expect(methods).to.not.have.property('prototype.testMethod');
+            expect(methods.testMethod).to.not.have.property('isStatic');
+            done();
+          });
+        });
+      });
+  });
+  
+  it('prototype method should not have isStatic property in method definition', function(done) {
+    ModelMethod.create(
+      {
+        modelId: userModel.id,
+        name: 'prototype.testMethod',
```

Comment:
Hmm, I think this is not valid usage of the API. The name should be always without `prototype.` prefix, at least that's what I thought?